var/spack/repos/builtin/packages/blis/package.py

```diff
@@ -0,0 +1,132 @@
+    available under a new/modified/3-clause BSD license. While BLIS exports a
+    new BLAS-like API, it also includes a BLAS compatibility layer which gives
+    application developers access to BLIS implementations via traditional BLAS
+    routine calls. An object-based API unique to BLIS is also available."""
+
+    homepage = "https://github.com/flame/blis"
+    url      = "https://github.com/flame/blis/archive/0.4.0.tar.gz"
+
+    version('0.4.0', sha256='9c7efd75365a833614c01b5adfba93210f869d92e7649e0b5d9edc93fc20ea76')
+    version('0.3.2', sha256='b87e42c73a06107d647a890cbf12855925777dc7124b0c7698b90c5effa7f58f')
+    version('0.3.1', sha256='957f28d47c5cf71ffc62ce8cc1277e17e44d305b1c2fa8506b0b55617a9f28e4')
+    version('0.3.0', sha256='d34d17df7bdc2be8771fe0b7f867109fd10437ac91e2a29000a4a23164c7f0da')
+    version('0.2.2', sha256='4a7ecb56034fb20e9d1d8b16e2ef587abbc3d30cb728e70629ca7e795a7998e8')
+
+    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
+
+    variant(
+        'threads', default='none',
+        description='Multithreading support',
+        values=('pthreads', 'openmp', 'none'),
+        multi=False
+    )
+
+    variant(
+        'blas', default=False,
+        description='BLAS compatibility',
+    )
+
+    variant(
+        'cblas', default=False,
+        description='CBLAS compatibility',
+    )
+
+    variant(
+        'shared', default=True,
+        description='Build shared library',
+    )
+
+    variant(
+        'static', default=False,
+        description='Build static library',
+    )
+
+    # TODO: add cpu variants. Currently using auto.
+    # If one knl, should the default be memkind ?
+
+    # BLIS has it's own API but can be made compatible with BLAS
+    # enabling CBLAS automatically enables BLAS.
+
+    provides('blas', when="+blas")
+    provides('blas', when="+cblas")
+
+    phases = ['configure', 'build', 'install']
+
+    def configure(self, spec, prefix):
+        config_args = []
+
+        config_args.append("--enable-threading=" + spec.variants['threads'].value)
+
+        if spec.variants['cblas']:
+            config_args.append("--enable-cblas")
+        else:
+            config_args.append("--disable-cblas")
+
+        if spec.variants['blas']:
+            config_args.append("--enable-blas")
+        else:
+            config_args.append("--disable-blas")
+
+        if self.variants['shared']:
+            config_args.append("--enable-shared")
+        else:
+            config_args.append("--disable-shared")
+
+        if self.variants['static']:
+            config_args.append("--enable-static")
+        else:
+            config_args.append("--disable-static")
+
+        configure("--prefix=" + spec.prefix,
+                  *config_args,
+                  "CC=" + env['CC'],
+                  "auto")
+
+    def build(self, spec, prefix):
+        make()
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def test(self):
```

Comment:
I would name the function `check` since that's what the target is. The name doesn't really matter anyway.